mlflow/models/cli.py

```diff
@@ -137,9 +137,9 @@ def build_docker(model_uri, name, install_mlflow):
 
 def _get_flavor_backend(model_uri, **kwargs):
     with TempDir() as tmp:
-        local_path = _download_artifact_from_uri(posixpath.join(model_uri, "MLmodel"),
+        local_path = _download_artifact_from_uri(posixpath.join(model_uri),
```

Comment:
Yep good point, this is due to a limitation in the models URI format, which currently allow URIs like
      - `models:/<model_name>/<model_version>`
      - `models:/<model_name>/<stage>`  (refers to the latest model version in the given stage)

But not `models:/<model_name>/<stage>/MLmodel` (there's no way to reference individual artifacts, like the "MLmodel" file, relative to the model). 

Supporting URIs like `models:/<model_name>/<stage>/MLmodel` seems reasonable to me, e.g. for use cases like this one, and I can't think of any systems-level limitations that would make it a bad idea (if you can download a full folder of model artifacts, you should be able to download a single file from that folder?). But want to confirm with @mparkhe @sueann before making that change :)

Reply:
Sounds good! 

I think that it would be great if models:/ behaved as other uris if possible, otherwise we need to treat them special or use only a subset of artifact repo apis / functionality.

If it's not possible, I think we should download the model first in and use local paths in the subsequent calls to avoid double download.

Reply:
Chatted offline with @sueann & @mparkhe, we may make extensions to the models URI format (e.g. adding namespaces / URIs like `models:/<namespace>/<model_name>/<stage>/`) that will make it difficult to disambiguate URIs like `models:/<model_name>/<stage>/MLmodel`, so for now just added logic to fetch the underlying model URI & get the MLmodel file from there